### DISCUSSION TITLE
Why does su not set `USER` and `LOGNAME` when target is `root`?

### DISCUSSION THREAD

**Discussion by septatrix** [util-linux/util-linux#1646] [Q&A] [answered]
I stumbled upon an issue which mistakenly assumed that both would always be updated but they are not. Looking in the man page and the code this seems to be on purpose so could someone please explain to me what the idea behind this is?

**Comment by karelzak** [comment, marked as answer]
It's original code from the coreutils project, and I don't remember the original purpose, but I guess it's a way how to provide info about the original user to the session, it just changes permissions and nothing else. The default is to do minimal changes to the environment. If you don't like it you need to use -l to create a more isolated session.
